language/tools/mirai-dataflow-analysis/README.md

```diff
@@ -0,0 +1,169 @@
+# MIRAI Dataflow Analyzer
+
+This tool (`mirai-dataflow`) provides an end-to-end method for running a dataflow analysis over a Rust crate.
+Internally, this tool uses [MIRAI](https://github.com/facebookexperimental/MIRAI) to generate a call graph of a Rust crate, as well as a Datalog
+representation of that call graph. Then, a dataflow analysis is run over the call graph using either
+[Differential Datalog](https://github.com/vmware/differential-datalog) or [Soufflé](https://souffle-lang.github.io/) and decoded results are presented.
+
+## Installation
+
+This tool depends on the presence of MIRAI, correctly installed on your system, as well as
+either an installation of either Differential Datalog or Soufflé Datalog. Thus, to get a complete
+setup, please follow the steps below:
+
+1. Install MIRAI by following the instructions in its [readme](https://github.com/facebookexperimental/MIRAI#readme).
+2. Install Soufflé and / or Differential Datalog.
+3. Set the nightly version of Rust for this crate using `rustup override set nightly-YYYY-MM-DD`.
+    - See [rust-toolchain](./rust_toolchain)) to determine the current compatible version, which in turn
+    must match the version [currently in use by MIRAI](https://github.com/facebookexperimental/MIRAI/blob/main/rust-toolchain).
+
+### Differential-Datalog-specific step
+
+If Differential Datalog is being used, please additionally use Differential Datalog to build `analyses/ddlog.dl`.
+```
+$ cd analyses/
+$ ddlog -i ddlog.dl
+$ cd ddlog_ddlog && cargo build --release
+```
+This will create a Rust crate (`analyses/ddlog_ddlog`) that includes a CLI that the `mirai-dataflow`
+interfaces with to perform its analysis.
+
+## Configuration
+
+`mirai-dataflow` is configured via a combination of CLI arguments and a configuration file.
+
+### CLI
+
+`mirai-dataflow` exposes the following CLI (viewable via the `--help` flag):
+```
+USAGE:
+    mirai-dataflow-analysis [FLAGS] [OPTIONS] <crate-path> <config-path>
```

Comment:
mirai-dataflow-analysis or just mirai-dataflow?

Reply:
Unfortunately that is automatically generated by structopt based on the crate name (which is mirai-dataflow-analysis).